sycl/source/detail/platform_impl.cpp

```diff
@@ -389,6 +390,19 @@ bool platform_impl::has(aspect Aspect) const {
 #include <CL/sycl/info/platform_traits.def>
 #undef __SYCL_PARAM_TRAITS_SPEC
 
+context platform_impl::getDefaultContext() {
+  const std::lock_guard<std::mutex> Guard(MContextMutex);
+
+  if (!MDefaultContexts.empty())
+    return MDefaultContexts.back();
+  else {
+    auto Devices = get_devices();
+    MDefaultContexts.emplace_back(Devices);
+
+    return MDefaultContexts.back();
+  }
```

Comment:
```suggestion
  auto Devices = get_devices();
  MDefaultContexts.emplace_back(Devices);

  return MDefaultContexts.back();
```
Suggest avoiding explicit else branch because if body has return statement.

Reply:
'k